### PR TITLE
Further tweak to is_in_or_equal

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

* Handle a corner case in is_in_or_equal
* Tweak is_running in move to cache check

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by abidlabs** [comment]
`Path.resolve()` can be a problem because it resolves symlinks, see: https://github.com/gradio-app/gradio/pull/3037

But I think in this case, its okay since we're only resolving the symlinks to check subpaths, we're not actually replacing filepaths with their resolved versions.

**Comment by freddyaboulton** [comment]
I think `.resolve` would be ok but just refactored to not resolve symlinks at all

File: gradio/utils.py [hunk 1]
```diff
@@ -113,6 +113,7 @@ def swap_blocks(self, demo: Blocks):
         # Copy over the blocks to get new components and events but
         # not a new queue
         demo._queue = self.running_app.blocks._queue
+        demo.is_launching = False
```
Comment (abidlabs) [inline gradio/utils.py:116]: Just to confirm, `demo.is_launching()` should already be False here, right?

**Review by abidlabs — APPROVED** [review]
lgtm @freddyaboulton. I went through the PR and everything makes sense to me. I just had one question above. 

Also, a nit, but I find the name `is_launching` a bit unclear (it sounds like its in the process of launching()`. I think renaming it `has_launched` and inverting it would be clearer. Feel free to change if you agree

**Comment by freddyaboulton** [comment]
Nice suggestion @abidlabs - just implemented and will merge once CI passes!

**Pull request merged**